tests/golden_tests/run_file/nat_add.hvm

```diff
@@ -0,0 +1,4 @@
+(Nat.add (Nat.zero)   x) = x
+(Nat.add (Nat.succ p) x) = (Nat.succ (Nat.add p x))
```

Comment:
Can you add a test to resugar an invalid Nat.succ like `(Nat.succ 0)` and a Nat.zero in an unexpected position like `(Nat.zero foo)`